src/Parallel/NodeLock.cpp

```diff
@@ -3,20 +3,25 @@
 
 #include "Parallel/NodeLock.hpp"
 
-#include <converse.h>
+#include <pup.h>
 
+#include "Parallel/Spinlock.hpp"
 #include "Utilities/ErrorHandling/Error.hpp"
 #include "Utilities/Gsl.hpp"
 
 namespace Parallel {
 
-NodeLock::NodeLock() : lock_(std::make_unique<CmiNodeLock>(CmiCreateLock())) {}
+NodeLock::NodeLock()
+    : lock_(std::make_unique<typename decltype(lock_)::element_type>()) {}
```

Comment:
[optional] Why go through the unique_ptr to get the type? Just use `SpinLock`? Same below.

Reply:
This makes it easier to switch the type if we want to in the future. I was going back and forth a bit trying to figure out what's best, including std::mutex, so only having to switch the type in the member variable was easier.